commands/migratecmd.go

```diff
@@ -250,18 +272,26 @@ func (mr *MigrateRunner) migrateObjs(cmObjs []object.ObjMetadata, reader io.Read
 	if err != nil {
 		return err
 	}
-	// Filter the ConfigMap inventory object.
-	rgInv, err := findResourceGroupInv(objs)
+	inv, objs, err := mr.rgLoader.InventoryInfo(objs)
 	if err != nil {
 		return err
 	}
-	rgInvClient, err := mr.rgProvider.InventoryClient()
-	if err != nil {
+
+	applier := apply.NewApplier(mr.rgProvider)
+
+	if err := applier.Initialize(); err != nil {
 		return err
 	}
-	inv := live.WrapInventoryInfoObj(rgInv)
-	_, err = rgInvClient.Merge(inv, cmObjs)
+	ch := applier.Run(context.TODO(), inv, objs, apply.Options{
```

Comment:
If there are local changes when we do this, is this still safe?

Reply:
Good question. Since the `objs` are loaded by `rgLoader` from the local not from the cluster, so the `applier.Run` applies whatever in the local.
If the change is for updating or adding resources, this is safe. If the change includes removing a resource, this will lead to the object being orphaned after the apply.

So we need to document there shouldn't be any local changes when calling the `migrate` command.

Reply:
So requiring that we update the actual resources as well as the inventory as part of the migration is unfortunate. A nice property of the existing migration path is that it only updates the inventory and does not touch any of the workloads. 

Would it be a better solution to just force users to do this in two steps? So instead of trying to update the `owning-inventory` annotation, we can just verify either that the resource containing the inventory information has the `managed-by` annotation or that all resources listed in the inventory list have the `owning-inventory` annotation? Fixing this for users should just be to run `apply`.

Reply:
I see what you mean. We should be able to do a better approach here. How about that the `migrate` command only updates the inventory object. What it does is to get the current inventory list from the cluster and then assign that list to ResourceGroup. So it doesn't bother with local changes since they are not applied to the cluster.

Then the `owning-inventory` annotation should be added by the next `apply`. We need to handle the pruning and applying correctly in the library. So that the first time apply after the migration doesn't break.

Reply:
The above approach covers the case where the previous apply was by an old version of `kpt`, where the `owning-inventory` was not added. 

If the previous apply was by the same version of `kpt` with ConfigMap inventory object, the above approach doesn't work. Since the resources already have the `owning-inventory` annotations that point to the ConfigMap. We need to fix these annotation. This can't be done by `apply`. 

So I think we need to update the `owning-inventory` in the resources from empty or the ConfigMap to the ResourceGroup CR. I'll update this PR for that.

Reply:
I tried to updating the `owning-inventory` annotation of objects in the cluster, but it is not a simple operation and we may not do it correctly. Here is the approach I used.
- read the list of object meta from the previous ConfigMap inventory object
- for each object meta, get the object from the cluster. If the object can be found, add the `owning-inventory` annotation to the object and update its `last-applied-configuration` annotation.

However the objects read from the cluster can't be used to run a re-apply. Or even the current update using a dynamic client passed, the next apply will fail.